benchmarks/src/main/scala/scalaz/zio/stacktracer/impls/AsmTracer.scala

```diff
@@ -0,0 +1,83 @@
+package scalaz.zio.stacktracer.impls
+
+import java.lang.invoke.SerializedLambda
+
+import org.objectweb.asm._
+import scalaz.zio.internal.stacktracer.Tracer
+import scalaz.zio.internal.stacktracer.ZTraceElement.SourceLocation
+import scalaz.zio.stacktracer.impls.AsmTracer.MethodSearchVisitor
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Java 8+ only
+ *
+ * file:line extraction will work for lambdas only.
+ * For subclasses of scala.Function1 it will point to the class definition
+ */
+final class AsmTracer extends Tracer {
+
+  final def traceLocation(lambda: AnyRef): SourceLocation = {
+    val clazz = lambda.getClass
+    val classLoader = clazz.getClassLoader
+
+    Try(clazz.getDeclaredMethod("writeReplace")) match {
```

Comment:
Why do you use a `Try` and not a `try catch` which seems more performant (less allocations) ?